### ISSUE TITLE
mpirun fails to show the error message

### ISSUE THREAD

**Issue by AboorvaDevarajan** [openpmix/prrte#1360]
### What version of the PMIx Reference Server are you using? (e.g., v1.0, v2.1, git master @ hash, etc.)
OMPI v5.0.x

```
$ git submodule status
 8c39d8e6a95d6fa78e765b5e86c324bd8a4ecd56 3rd-party/openpmix (v4.1.2-58-g8c39d8e6)
 f75647a0518b5a476011f543200fca1cf8600cb8 3rd-party/prrte (v2.0.2-99-gf75647a051)
$ git branch
* v5.0.x
```
### Please describe the system on which you are running

* Operating system/version: RH8.6
* Computer hardware: ppc64le
* Network type:

-----------------------------

## Details of the problem

```
[prrte]$ mpicc hello_world.c -o hello_world
[prrte]$ mpirun -np 2 --map-by core --map-by core ./hello_world
[prrte]$ echo $?
213
```

The expected behaviour seems to be that we should print the error message before exiting, but looks like the error message is not printed,
https://github.com/openpmix/prrte/blob/f75647a0518b5a476011f543200fca1cf8600cb8/src/mca/schizo/base/schizo_base_frame.c#L403

### Cause:

The error message is only printed if PMIx is initialized, (PMIx_Init) otherwise it just returns, so if some error occurs before PMIx is initialized prte just returns without printing the error message. 

Code path to look for,

```
prte_show_help
prte_show_help_norender
PMIx_Log_nb  (returns if init_cntr <= 0)
```

**Comment by rhc54** [comment]
It's fixed on the master branch - just hasn't come back to the release branch and/or made it over to OMPI.

**Comment by rhc54** [comment]
Verified this is now fixed on the PRRTE v2.1 branch - probably needs updating of submodule pointer on OMPI.
